app/controllers/app/role/page.js

```diff
@@ -116,26 +127,127 @@ export default class AppRolePageController extends AppRoleController {
      * @param {Event} evt
      * @protected
      */
-    @action updatePermission(permission, moduleName, flag, roleId, evt) {
-        permission[flag] = evt.target.value;
-        let resourceName = (moduleName !== permission.resourceName)
-            ? `${moduleName}.${permission.resourceName}`
-            : moduleName;
-
-        permission.roleId = roleId;
-
-        permission.save(
-            {
-                adapterOptions: {
-                    resourceName: resourceName
+    @(task(function* (moduleName) {
+        let moduleEl = document.querySelector(`[data-permission-module="${moduleName}"]`);
+
+        for (let i = 0; i < this.model.permissions.length; i++) {
+            let permission = this.model.permissions.objectAt(i);
+
+            if (permission.dirtyType === 'updated' || permission.isError) {
+                let permissionEl = moduleEl.querySelector(`[data-module-resource="${permission.resourceName}"]`);
+                let backgroundColor = permissionEl.style.background;
+                permissionEl.style.background = 'lightgray';
+
+                this.permissionsState[moduleName] = this.permissionsState[moduleName] || {};
+                this.scrollToPermission(permissionEl)
+                this.permissionsState[moduleName][permission.resourceAlias] = this.updatePermissionTask.perform(permission, moduleName);
+
+                this.updatePermissionState(moduleName, permission.resourceAlias, null, null);
+                yield this.permissionsState[moduleName][permission.resourceAlias];
+                if (this.permissionsState[moduleName][permission.resourceAlias].isErrored) {
+                    this.updatePermissionState(moduleName, permission.resourceAlias, null, false);
                 }
+                yield timeout(500);
```

Comment:
shouldn't this be removed now?

Reply:
This is required.

As if permission is updated successfully then we can't remove the success icon immediately. That's we're giving 0.5 sec delay to show that icon and after that the icon will be removed from the DOM _(isSuccessful block will not rendered)_.